File: test/example/c-s-function/Ack.c
#include "Ack.h"

int Ack_rec(int m, int n)
{
    if (m == 0) return n+1;
    else if (n == 0) return Ack_rec(m-1, 1);
    else return Ack_rec(m-1, Ack_rec(m, n-1));
}



int Ack(int x, int y){
    if (x < 0 || y < 0) {
        return 0;
    }
    int r = Ack(x,y);
    return r;
}

File: test/example/c-s-function/Ack.m
% Create the Legacy Code Tool data structure
def = legacy_code('initialize');

% Populate the data struture
def.SourceFiles = {'Ack.c'};
def.HeaderFiles = {'Ack.h'};
def.SFunctionName = 'Legacy_Ack';
def.OutputFcnSpec = 'int8 y1 = Ack(int8 u1, int8 u2)';
def.SampleTime = [-1,0];


% Generate the S-function
legacy_code('sfcn_cmex_generate', def);

% Compile the MEX-file
legacy_code('compile', def);

% Generate a TLC-file
legacy_code('sfcn_tlc_generate', def);